### ISSUE TITLE
Generation of complex cells not working

### ISSUE THREAD

**Issue by kcaisley** [patrickschulz/openPCells#15]
Hi there, really cool project!

I've built this project with gcc 13.2 on Fedora 39, and gcc 11.4 on RHEL9. In both cases, I tried installing system-wide via `./configure.sh && make && sudo make install`as well as doing a local install without sudo permissions: `./configure --all-load-paths-local && make`.

The projects appears to build properly with only a couple warnings of the `switch`, `unused-function`, and `maybe-uninitialized` variety.

I'm able to run the basic transistor pcell, which does produce a svg/gds file:

```
$ ./opc --technology opc --export svg --cell basic/mosfet fingers=4
export_write_write_toplevel: could not initialize export
this export does not know how to write hierarchies, hence the cell is being written flat
```

But when I try to generate something more complicated, the tool appears to not execute properly. A couple examples:

```
$ ./opc --technology opc --export gds --cell basic/mosfet gatelength=300 gatespace=500 fwidth=1500 fingers=4 drawtopgate=true connectsource=true drawguardring=true guardringxsep=200 guardringysep=200
modules/pcell.lua:494: argument 'guardringxsep' has no matching parameter in cell 'basic/mosfet', maybe it was spelled wrong?
stack traceback:
	[C]: in function 'error'
	modules/pcell.lua:494: in field 'evaluate_parameters'
	scripts/create_cell.lua:63: in main chunk
errors while creating cell


$ ./opc --technology opc --export gds --cell analog/current_starved_ringoscillator
modules/pcell.lua:321: trying to overwrite parameter 'outergstwidth', which does not exist in cell 'basic/cmos'
stack traceback:
	[C]: in function 'assert'
	modules/pcell.lua:321: in function <modules/pcell.lua:308>
	(...tail calls...)
	cell 'analog/current_starved_ringoscillator':110: in field 'layout'
	modules/pcell.lua:536: in upvalue '_create_layout_internal'
	modules/pcell.lua:572: in field 'create_layout_env'
	scripts/create_cell.lua:67: in main chunk
errors while creating cell


$ ./opc --technology opc --export gds --cell stdcells/not_gate
modules/pcell.lua:239: argument 'gstspace' has no matching parameter in cell 'basic/cmos', maybe it was spelled wrong?
stack traceback:
	[C]: in function 'assert'
	modules/pcell.lua:239: in upvalue '_get_parameters'
	modules/pcell.lua:522: in upvalue '_create_layout_internal'
	modules/pcell.lua:554: in field 'create_layout'
	cell 'stdcells/harness':41: in field 'layout'
	modules/pcell.lua:536: in upvalue '_create_layout_internal'
	modules/pcell.lua:554: in field 'create_layout'
	cell 'stdcells/not_gate':35: in field 'layout'
	modules/pcell.lua:536: in upvalue '_create_layout_internal'
	modules/pcell.lua:572: in field 'create_layout_env'
	scripts/create_cell.lua:67: in main chunk
errors while creating cell
```

I've never debugged Lua before, but it seems like certain parameters just aren't being found. Perhaps something moved recently in the project organization which broken this?

**Comment by patrickschulz** [comment]
Hi and thanks for the report. Those are definitely bugs due to slightly outdated documentation and change of parameter names. I fixed the obvious issues in 826290eda4cbbd3f822e7c126c0796ddf1e999fc. Unfortunately a lot of stuff has changed in the last month and I haven't been keeping up with the documentation and cells I'm not currently using. I'm also working on a revised cell system (which will possibly obsolete the --cell option), but that might take a while. Until then I won't be able to find more subtle bugs in the cells. But if you encounter any please let me know, as it's usually quite fast to fix them.

**Comment by kcaisley** [comment]
Excellent, thank you!
